### PR TITLE
Adding validation rules, Bikram Sambat date and date-part types

### LINKED ISSUE #611: Add documentation for new elements added to support reporting exact LMP date
Related to medic/cht-core#4613
With medic/cht-core#7422, two new validation rules have been added:
1. `isBefore(duration)`
2. `isAfter(duration)`

Also, field types were added for SMS forms to support Bikram Sambat dates:
1. `bsDate`
2. `bsYear`
3. `bsMonth`
4. `bsDay`

These need to be documented.

### PR DESCRIPTION
**Pull request by binokaryg**
Closes #611

### CODE REVIEW

**Comment by mrjones-plip** [comment]
@ngaruko - can I impose on you to review the docs PR? I figure since you just did AT for [the ticket](https://github.com/medic/cht-core/issues/4613), this should be pretty straight forward.

@binokaryg - are there any of these items that are are only available >3.15.0 given that is when this feature will ship?

File: content/en/apps/reference/app-settings/_index.md [hunk 1]
```diff
@@ -224,3 +224,5 @@ matches "mary", "Mary", "john", "John", "JOhN", etc. Not "maryjohn"
...
 | `isISOWeek(weekFieldName[, yearFieldName])` | Returns `true` if the current report has a week value that is less or equal to the number of ISO weeks of the current year or the year value of the same report. The first parameter is the field name for the week and the second parameter is the field name for the year: `isISOWeek('week', 'year')`. If the second parameter is not specified, then the current year is used: `isISOWeek('week')`. |
+| `isAfter(duration)` | Returns `true` if the date property comes after today plus the duration. For this to work, the property should be a date type. e.g. `isAfter('2 weeks')` checks that the date property is at least two weeks in future from today. Negative values are also supported. |
```
Comment (ngaruko) [inline content/en/apps/reference/app-settings/_index.md:227]: Looks good. If you feel like (sounds like a silly nitpick), you could add a `the`  before `future` (`in future`), but ... :)

**Review by ngaruko — APPROVED** [review]
All good. Just a silly grammar nitpick. 😄

**Review by ngaruko — APPROVED** [review]
Great!

**Pull request merged**